### DISCUSSION TITLE
Any way to use a Lua package like socket.http in extension?

### DISCUSSION THREAD

**Discussion by dmonder** [quarto-dev/quarto-cli#8056] [Q&A]
### Description

I would like to create a short code to include some text from a web API call but can't seem to figure out how to load the Lua socket.http package in Quarto. Is this possible? If so, how do I do it.

Right now, I have the following as the first line in my Lua extension:

`local http = require("socket.http")`

but this generates the following error:

```
Error running filter C:/Program Files/Quarto/share/filters/main.lua:
[string "..."]:295: module 'socket.http' not found:
	no field package.preload['socket.http']
```

**Comment by cderv** [comment]
The Lua module needs to be available for your extension. Did you try to bundle `socket.http` in your Extensions ? I believe [require](https://www.lua.org/pil/8.1.html) in Lua knows how to load a standalone library provide as a file. 

More generaly, the feature you want is part of LuaSocket and must be made available somehow (https://w3.impa.br/~diego/software/luasocket/home.html#download) as this is not part of available Lua API provided by Pandoc or Quarto (https://quarto.org/docs/extensions/lua-api.html) 

So I am not sure how, but it needs to be made available for the Lua interpreter.

**Reply by dmonder** [reply]
I am looking into the LuaSocket source. I did discover that it is at [https://lunarmodules.github.io/luasocket/](https://lunarmodules.github.io/luasocket/) now.

**Comment by nickvigilante** [comment]
I had this same question, and I'm not sure if this is useful to you, but my colleague told me about [`pandoc.mediabag.fetch`](https://pandoc.org/lua-filters.html#pandoc.mediabag.fetch), which returns the MIME type (or nil) and the response body (or nil) from the URL in question. I tested this out on my local install, and it works like a charm.
